src/index.mjs

```diff
@@ -118,11 +125,20 @@ function astroRehypeRelativeMarkdownLinks(options = {}) {
       // NOTE - When there is a content collection name being applied, this only handles situations where the physical
       //        directory name of the content collection maps 1:1 to the site page path serviing the content collection
       //        page (see details above)
-      let webPathFinal = [
-        collectionName === "" ? "" : path.posix.sep + collectionName,
+      const resolvedUrl = [
+        collectionName === NO_COLLECTION_NAME
+          ? ""
+          : path.posix.sep + collectionName,
         resolvedSlug,
       ].join(path.posix.sep);
 
+      // slug of empty string ('') is a special case in Astro for root page (e.g., index.md) of a collection
+      let webPathFinal = applyTrailingSlash(
+        (frontmatterSlug === "" ? "/" : frontmatterSlug) || url,
```

Comment:
What do you think about doing the same thing here that was done for `NO_COLLECTION_NAME`?

Reply:
lol, I never did like the way I wrote that code where the collectionName would be `''` when `collectionPathMode` was `root` - the collection name is still the collection name, it's just whether or not we use it in the path :)  Just pushed a commit that I'm happier with and accomplishes both of our goals I think.  Lemme know!

Reply:
Looks great! Thanks!